providers/storage/inmemoryDocStore.js

```diff
@@ -60,12 +60,12 @@ class InmemoryDocStore {
     }));
   }
 
-  delete(type, url) {
+  delete(type, key) {
     const collection = this.collections[type];
     if (!collection) {
       return Q(null);
     }
-    delete collection[url];
+    delete collection[key];
```

Comment:
delete both the entries for the url and urn.  Look in the document (if any) to get the key you do not have.